src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -91,256 +439,367 @@ private void StopMonitor()
-                    {
-                        sb.Append(CounterProvider.SerializeUnknownProviderName(providerName, _interval));
-                    }
-                    else
-                    {
-                        sb.Append(provider.ToProviderString(_interval));    
-                    }
-                    
-                    if (i != _counterList.Count - 1)
-                    {
-                        sb.Append(",");
-                    }
-
-                    if (tokens.Length == 1)
+                    if(counterNames.IndexOf(']') == -1)
                     {
-                        filter.AddFilter(providerName); // This means no counter filter was specified.
+                        throw new FormatException("Expected to find closing ']' in counter_provider");
                     }
                     else
                     {
-                        string counterNames = tokens[1];
-                        string[] enabledCounters = counterNames.Substring(0, counterNames.Length-1).Split(',');
-                        
-                        filter.AddFilter(providerName, enabledCounters);
+                        throw new FormatException("Unexpected characters after closing ']' in counter_provider");
                     }
                 }
-                providerString = sb.ToString();
+                string[] enabledCounters = counterNames.Substring(0, counterNames.Length - 1).Split(',', StringSplitOptions.RemoveEmptyEntries);
+                counters.AddProviderCounters(providerName, enabledCounters);
             }
-            return providerString;
         }
 
-        private async Task<int> Start()
+        private EventPipeProvider[] GetEventPipeProviders()
         {
-            string providerString = BuildProviderString();
-            if (providerString.Length == 0)
+            // EventSources support EventCounter based metrics directly
+            IEnumerable<EventPipeProvider> eventCounterProviders = _counterList.Providers.Select(
+                providerName => new EventPipeProvider(providerName, EventLevel.Error, 0, new Dictionary<string, string>()
+                {{ "EventCounterIntervalSec", _interval.ToString() }}));
+
+            //System.Diagnostics.Metrics EventSource supports the new Meter/Instrument APIs
+            const long TimeSeriesValues = 0x2;
+            StringBuilder metrics = new StringBuilder();
+            foreach(string provider in _counterList.Providers)
             {
-                return ReturnCode.ArgumentError;
+                if(metrics.Length != 0)
+                {
+                    metrics.Append(",");
+                }
+                if(_counterList.IncludesAllCounters(provider))
+                {
+                    metrics.Append(provider);
+                }
+                else
+                {
+                    string[] providerCounters = _counterList.GetCounters(provider).Select(c => "{provider}\\{counter}").ToArray();
```

Comment:
Ah, this explains why I couldn't filter metrics with the provider syntax :) 
```suggestion
                    string[] providerCounters = _counterList.GetCounters(provider).Select(counter => $"{provider}\\{counter}").ToArray();
```

Reply:
You can filter counters using syntax like `System.Runtime[cpu-usage,working-set]`. This PR changed the help text to hopefully make that syntax better understood but it shouldn't have changed what the tool does/doesn't accept.

Were you saying that some filtering that used to work in dotnet-counters no longer does?

Reply:
Without this fix your string was the literal string "provider\counter", and I couldn't filter particular instruments with the usual syntax. Once I added this fix I was able to.

Reply:
ah sorry, I completely misinterpreted you above. Not sure why, in hindsight what you are saying is very clear : ) And yep, definitely a bug I should fix!